bftengine/src/bftengine/messages/ReplicaAsksToLeaveViewMsg.hpp

```diff
@@ -25,7 +25,7 @@ namespace impl {
 
 class ReplicaAsksToLeaveViewMsg : public MessageBase {
  public:
-  enum class Reason : uint8_t { ClientRequestTimeout };
+  enum class Reason : uint8_t { ClientRequestTimeout, PrimaryGetInChargeTimeout };
```

Comment:
NewPrimaryGetInChargeTimeout ?

Reply:
I agree. This will better indicate that the expected New Primary is given a time period to take action.